README.md

```diff
@@ -1,7 +1,69 @@
-# sdk-php
+# PHP SDK for [CloudEvents](https://github.com/cloudevents/spec)
+
+## Status
+
+This SDK is currently a work in progress, therefore things might (and will) break with every update.
+
+This SDK aims to supports the following versions of CloudEvents:
+
+- [v1.0](https://github.com/cloudevents/spec/blob/v1.0.1/spec.md)
+
+## Installation
+
+Install the SDK using [composer](https://getcomposer.org/):
+```sh
+composer install cloudevents/php-sdk
+```
+
+## Send your first CloudEvent
+
+```php
+use \CloudEvents\Client;
+use \CloudEvents\Event;
+use \CloudEvents\Request;
+
+$event = (new Event())
+    // The current implementation requires you to maintain your own id.
+    ->setId('1n6bFxDMHZFChlI4TVI9tdzphB9')
+    ->setSource('/examples/php-sdk')
+    ->setType('com.example.type')
+    ->setData(json_encode(['example' => 'first-event']));
+
+(new Client('http://localhost:8080/'))
+    ->sendRequest(new Request($event));
+```
+
+Note that the `CloudEvents\Client` implements the [PSR-18](https://www.php-fig.org/psr/psr-18/) spec and the `CloudEvents\Request` implements the appropriate [PSR-7 interfaces](https://www.php-fig.org/psr/psr-7/).
+
+## Serialize/Deserialize a CloudEvent
```

Comment:
It would be most useful to show an example of **parsing/consuming** a JSON string into a `CloudEvent` PHP object. That's how most developers will use CloudEvents.

Reply:
Agreed! This is just a point of ideation to get us started so this README could be much more valuable. These methods don't even exist yet, so I'm just trying to make a mental note of providing an example post implementation here.

Reply:
I think this is a good starting off point to get the discussion going and start building out some kind of implementation.